arch/arm/core/irq_manage.c

```diff
@@ -188,3 +188,65 @@ void _arch_isr_direct_header(void)
 }
 #endif
 
+#if defined(CONFIG_ARM_SECURE_FIRMWARE)
+/**
+ *
+ * @brief Set the target security state for the given IRQ
+ *
+ * Function sets the security state (Secure or Non-Secure) targeted
+ * by the given irq. It requires ARMv8-M MCU.
+ * It is only compiled if ARM_SECURE_FIRMWARE is defined.
+ * It should only be called while in Secure state, otherwise, a write attempt
+ * to NVIC.ITNS register is write-ignored(WI), as the ITNS register is not
+ * banked between security states and, therefore, has no Non-Secure instance.
+ *
+ * It shall assert if the operation is not performed successfully.
+ *
+ * @param irq IRQ line
+ * @param secure_state 1 if target state is Secure, 0 otherwise.
+ *
+ * @return N/A
+ */
+void irq_target_state_set(unsigned int irq, int secure_state)
+{
+	if (NVIC_GetTargetState(irq) == 0) {
```

Comment:
Why we have to check current state of the IRQ?
Is there any HW limitation that prevents us to set the same state twice?

Reply:
Not that I know of. Then, I can simplify this code, I suppose, removing the check on the current state.

Reply:
IMHO we should use simpler approach.